ipamd/ipamd.go

```diff
@@ -420,6 +423,27 @@ func (c *IPAMContext) getLocalPodsWithRetry() ([]*k8sapi.K8SPodInfo, error) {
 		return nil, nil
 	}
 
+	var containers map[string]*cri.ContainerInfo
```

Comment:
Might be nice to put a comment above this block that explains why you're doing this. Something like this might work...

```
// Ask CRI for the locally-running containers and attempt to set the
// Pod.Container field to the value of the *sandbox* container. This
// will ensure that we don't unassign IPs from Pods due to looking up
// Pods with the wrong container ID.
```

Reply:
Will do -- should hopefully be clearer once the sandbox rename is done, too.

Reply:
Worth noting that this code was largely pre-existing -- I restored it in the first commit with a revert, and modified it to use the `cri` package instead of `docker`, but didn't otherwise touch it. (e.g. the TODO was there before).

Added a comment and tidied up the other comment to be a bit clearer.